src/libcore/iter/adapters/mod.rs

```diff
@@ -1615,9 +1615,8 @@ impl<I> Iterator for Skip<I> where I: Iterator {
             let to_skip = self.n;
             self.n = 0;
             // nth(n) skips n+1
-            if self.iter.nth(to_skip-1).is_none() {
-                return None;
-            }
+            self.iter.nth(to_skip - 1).as_ref()?;
```

Comment:
I think this should not be done in this PR. Using `?` may have performance implications compared to the old code (as well as creating the reference -- which seems odd).

Reply:
I honestly have no idea about this `as_ref()`, possibly it was added by IDE when I was applying lints.

`?` has been used 14 times in this whole file although it performs a bit worse: https://godbolt.org/z/nHS4fh
I'll drop it but what about other 14 uses in this file, should they be left as is or replaced?